lecture-notes/4-cor.md

```diff
@@ -139,32 +174,51 @@ Gary Bernhardt.
 ## Lecture 2
 
 Even simple software has edge cases. In CS 15, for example, one assignment
-involves writing a `delete` function to remove an element from a binary search
-tree and maintain the BST invariant. This function alone had several cases: the
-node is `NULL`; the node has no children; the node has one child; the node has
+involves writing a `remove` function to remove an element from a binary search
+tree and maintain the BST invariant.
+
+```c++
+Node* BST::remove(Node* root, int value) {
+  // ...
+}
+```
+
+This function alone had several cases: the node is `NULL`; the root has the
```

Comment:
`s/had several cases/must tolerate several different inputs/` or something? I think we should be more precise about what we mean by a "case".